test/unit/navigationValidation.test.js

```diff
@@ -0,0 +1,136 @@
+import { isPanelEnabled } from "../../src/helpers/navigationValidation.js";
+
+const initialState = {
+  data: [],
+  selectedFeatures: [],
+  accuracyCheckExamples: [],
+  saveStatus: "notStarted",
+  trainedModelDetails: {}
+};
+
+const dataUploadedState = {
+  ...initialState,
+  data: [
+    {
+      name: "Hermione",
+      isEvil: false,
+      house: "Gryffindor"
+    },
+    {
+      name: "Harry",
+      isEvil: false,
+      house: "Gryffindor"
+    },
+    {
+      name: "Voldemort",
+      isEvil: true,
+      house: "Slytherin"
+    }
+  ]
```

Comment:
I wonder if there's also a Tom Riddle whose `isEvil` state we don't know until the end of the app? :)